### PR TITLE
jellyfin-mpv-shim: 1.7.1 -> 1.10.1, enable desktop mode

### PR DESCRIPTION
**Pull request by jojosch**
###### Motivation for this change

Update to [latest upstream release](https://github.com/iwalton3/jellyfin-mpv-shim/releases/tag/v1.10.0). Also enabled the desktop version of the MPV shim.

Supersedes #108989.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
```
/nix/store/isfmqkbdw0fgwfac6sjbfx80cpd5gc1q-jellyfin-mpv-shim-1.7.1	  477168640
/nix/store/hlskj8yjwys8dznqkaiw3wmqyx08p2zj-jellyfin-mpv-shim-1.10.1	  963861152
```
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/video/jellyfin-mpv-shim/default.nix [hunk 1]
```diff
@@ -34,24 +73,22 @@ buildPythonApplication rec {
...
 
-  propagatedBuildInputs = [
-    jellyfin-apiclient-python
-    mpv
-    pillow
-    python-mpv-jsonipc
+  postInstall = ''
+    # Create desktop item.
+    mkdir -p $out/share/applications
+    cp ${desktopItem}/share/applications/* $out/share/applications
```
Comment (SuperSandro2000) [inline pkgs/applications/video/jellyfin-mpv-shim/default.nix:83]: Please use copyDesktopItems

File: pkgs/applications/video/jellyfin-mpv-shim/default.nix [hunk 2]
```diff
@@ -34,24 +73,22 @@ buildPythonApplication rec {
   '';
 
   postPatch = ''
-    # link the default shader pack
-    ln -s ${shaderPack} jellyfin_mpv_shim/default_shader_pack
+    sed -i 's~"check_updates":\s*True,~"check_updates": False,~' jellyfin_mpv_shim/conf.py
+    sed -i 's~"notify_updates":\s*True,~"notify_updates": False,~' jellyfin_mpv_shim/conf.py
```
Comment (SuperSandro2000) [inline pkgs/applications/video/jellyfin-mpv-shim/default.nix:77]: Please use substituteInPlace

File: pkgs/applications/video/jellyfin-mpv-shim/default.nix [hunk 3]
```diff
@@ -1,25 +1,64 @@
...
+    pywebview
+
+    # desktop dependencies
+    flask
+    pyqtwebengine
+    werkzeug
+  ];
+
+  nativeBuildInputs = [
+    pyqtwebengine.wrapQtAppsHook
```
Comment (SuperSandro2000) [inline pkgs/applications/video/jellyfin-mpv-shim/default.nix:61]:
```suggestion
    qt5.wrapQtAppsHook
```

**Comment by jojosch** [comment]
@SuperSandro2000 now using `substituteInPlace`, `copyDesktopItems` and `qt5.wrapQtAppsHook`

**Comment by jojosch** [comment]
There was a new release today (https://github.com/jellyfin/jellyfin-desktop/releases/tag/v1.9.0) and the MPV Shim is now a official Jellyfin project.

**Comment by jojosch** [comment]
Updated to latest release https://github.com/jellyfin/jellyfin-desktop/releases/tag/v1.10.0

**Comment by jojosch** [comment]
And another release https://github.com/jellyfin/jellyfin-desktop/releases/tag/v1.10.1

**Pull request merged**